src/constructs/dns/dns-records.ts

```diff
@@ -0,0 +1,57 @@
+import { CfnResource } from "@aws-cdk/core";
+import type { GuStack } from "../core";
+
+export enum RecordType {
+  CNAME = "CNAME",
+}
+
+export interface GuDnsRecordSetProps {
+  name: string;
+  recordType: RecordType;
+  resourceRecords: string[];
+  ttl?: number;
+}
+
+/**
+ * Do not use this construct directly. Use [[`GuCname`]] instead in order to reduce boilerplate.
+ */
+export class GuDnsRecordSet {
+  constructor(scope: GuStack, id: string, props: GuDnsRecordSetProps) {
+    // The spec for this private resource type can be found here:
+    // https://github.com/guardian/cfn-private-resource-types/tree/main/dns/guardian-dns-record-set-type/docs#syntax
+    new CfnResource(scope, id, {
+      type: "Guardian::DNS::RecordSet",
+      properties: {
+        Name: props.name,
+        ResourceRecords: props.resourceRecords,
+        RecordType: props.recordType,
+        TTL: props.ttl ?? 3600,
+        Stage: scope.stage,
+      },
+    });
+  }
+}
+
+export interface GuCnameProps {
+  /** The name of the record, for example: xyz.example.com */
+  name: string;
+  /** The record your CNAME should point to, for example your Load Balancer DNS name */
+  resourceRecord: string;
```

Comment:
Since you can only have one `CNAME` for a given name, we use different props to prevent users from making mistakes. An alternative would be to throw an error inside `GuDnsRecordSet`, but preventing these mistakes via the type system felt more appropriate to me.